text/chapter5.md

```diff
@@ -426,66 +381,80 @@ So, constructing values of algebraic data types is simple, but how do we use the
 
 ```haskell
-showPoint :: Point -> String
-showPoint (Point { x, y }) = ...
+{{#include ../exercises/chapter5/src/ChapterExamples.purs:electricalUnits}}
 ```
 
-Here, we only specify the names of the properties, and we do not need to specify the names of the values we want to introduce. This is called a _record pun_.
-
-It is also possible to use record puns to _construct_ records. For example, if we have values named `x` and `y` in scope, we can construct a `Point` using `Point { x, y }`:
+Then we define functions and values using these types:
 
 ```haskell
-origin :: Point
-origin = Point { x, y }
-  where
-    x = 0.0
-    y = 0.0
+{{#include ../exercises/chapter5/src/ChapterExamples.purs:calculateCurrent}}
 ```
 
-This can be useful for improving readability of code in some circumstances.
+This prevents us from making silly mistakes, such as attempting to calculate the current produced by _two_ lightbulbs _without_ a voltage source.
 
-## Exercises
+```haskell
+current :: Amp
+current = calculateCurrent lightbulb lightbulb
+{-
+TypesDoNotUnify:
+  current = calculateCurrent lightbulb lightbulb
+                             ^^^^^^^^^
+  Could not match type
+    Ohm
+  with type
+    Volt
+-}
+```
 
-1. (Easy) Write a function `circleAtOrigin` which constructs a `Circle` (of type `Shape`) centered at the origin with radius `10.0`.
-1. (Medium) Write a function `doubleScaleAndCenter` which scales the size of a `Shape` by a factor of `2.0` and centers it at the origin.
-1. (Medium) Write a function `shapeText` which extracts the text from a `Shape`. It should return `Maybe String`, and use the `Nothing` constructor if the input is not constructed using `Text`.
+If we instead just used `Number` without `newtype`, then the compiler can't help us catch this mistake:
 
-## Newtypes
+```haskell
+-- This also compiles, but is not as type safe.
+calculateCurrent :: Number -> Number -> Number
+calculateCurrent v r = v / r
+
+battery :: Number
+battery = 1.5
 
-There is an important special case of algebraic data types, called _newtypes_. Newtypes are introduced using the `newtype` keyword instead of the `data` keyword.
+lightbulb :: Number
+lightbulb = 500.0
 
-Newtypes must define _exactly one_ constructor, and that constructor must take _exactly one_ argument. That is, a newtype gives a new name to an existing type. In fact, the values of a newtype have the same runtime representation as the underlying type. They are, however, distinct from the point of view of the type system. This gives an extra layer of type safety.
+current :: Number
+current = calculateCurrent lightbulb lightbulb -- uncaught mistake
+```
 
-As an example, we might want to define newtypes as type-level aliases for `Number`, to ascribe units like pixels and inches:
+This design principle is perhaps better [communicated visually](https://twitter.com/jusrin00/status/875238742621028355/photo/1).
```

Comment:
Could be a slippery slope with non-technical links, but I think this is too perfect not to include.